streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -581,6 +652,24 @@
                     1000,
                     Importance.LOW,
                     BUFFERED_RECORDS_PER_PARTITION_DOC)
+            .define(BUILT_IN_METRICS_VERSION_CONFIG,
+                    Type.STRING,
+                    METRICS_24,
+                    in(
+                        METRICS_0100,
+                        METRICS_0101,
+                        METRICS_0102,
+                        METRICS_0110,
+                        METRICS_10,
+                        METRICS_11,
+                        METRICS_20,
+                        METRICS_21,
+                        METRICS_22,
+                        METRICS_23,
+                        METRICS_24
+                    ),
```

Comment:
I implemented this as defined in KIP-444. Now I am a bit concerned about the maintenance of future versions here. Thus, I would like to propose the following:
- use values `current`, `before 2.4` (or `<2.4`, or `0.10.0-2.3`). 
- each time a change is added that breaks compatibility, we would add a value `between 2.4 and x.y` (or `>=2.4, <x.y`, or `2.4-(x.y-1))`).
- default value would always stay `current`
WDYT?

Reply:
In addition the test burden increases, because in integration tests for metrics we need to test all ten versions.

Reply:
+100

Reply:
Yeah, this is crazy. Also, think of the burden on users: we shouldn't offer different configs that result in no differences.

Let's revisit the KIP discussion. We should only offer `current` (or `latest` is a common version "number" for this concept), and then only add version ranges after the fact. As you say, we're breaking compatibility in 2.4, so we need to add a `0.10-2.3` option to pin older configs. If we break it again in 7.0, we'd add `2.4-6.8`.